### ISSUE TITLE
How to implement saving

### ISSUE THREAD

**Issue by Noah12317** [AlisterT/openjazz#92]
all you need to do to save is to save variables to a file as what level the player is in, and any other things (it only seems to be what level the player is in though, not position, not live, not health, but maybe other parts save things) but it wouldnt be to difficult, it may only take up to 10 lines of code

**Comment by carstene1ns** [comment]
That is right. However we actually want to be compatible with original save games instead of creating a custom format.

Unfortunately the format is obfuscated.

**Comment by Noah12317** [comment]
> That is right. However we actually want to be compatible with original save games instead of creating a custom format.
> 
> Unfortunately the format is obfuscated.

well i doubt anyone is that far into the game to switch builds of the game entirely (will most likely feel its risky and files may get damaged) and its better than nothing. also it will most likely take a long time until anyone figures out the save date encryption, so it should be added at least until the real save data encryption can be used. also its been 3 years since the last update

**Comment by Violet-CLM** [comment]
[CYBERDEV](https://www.jazz2online.com/jcf/showthread.php?p=495372#post495372) has [made some progress on save files](https://codeberg.org/CYBERDEV/JJSave). (I'm just posting this link because I have a github account and they don't.)

**Comment by carstene1ns** [comment]
I have started working on this. Loading games from the main menu is now possible. 🎉

**Comment by billbeans** [comment]
still no support for saving?

**Comment by carstene1ns** [comment]
Well, I tried to use the algorithm for saving, but the original game does not accept the files, so one detail is still missing.
I can add the current implementation, but the savegames are only loaded by OJ then.

**Comment by billbeans** [comment]
> Well, I tried to use the algorithm for saving, but the original game does not accept the files, so one detail is still missing. I can add the current implementation, but the savegames are only loaded by OJ then.

I feel like that would be preferable at this point. OJ plays pretty true to the original with default settings. I would probably skip DOSBox altogether for my next playthrough. Compatible saves would be nice though, I get the frustration.

**Comment by carstene1ns** [comment]
Fixed by f2aed10f444180734e12b374807bbae16f5185d3. Though not all fields are saved yet.
